### ISSUE TITLE
Change sort order in list view

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/NGT-signbank#297]
* Glosses that start with special characters appear at the top now. Please change the sort order so that we first see A-Z, then all other characters.
* For both ASL and NGT

**Comment by henrinie** [comment]
This is dependent on the database collation, so you need to change that for the fields you want to sort differently (if not all).

**Comment by Woseseltops** [comment]
@henrinie is right, this is database related... I'm going to have to dive deep into the database, as Django has no functionality to get this right

**Comment by henrinie** [comment]
@Woseseltops If I remember correctly, it is somehow possible to add the collation to each query django makes. But it might be a better idea to edit the collation directly on the database.

**Comment by Woseseltops** [comment]
This turned out to be quite a technical challenge: 
* At first I tried to do it in the database, as @henrinie suggested, but after a time consuming Google search and a lot of experimentation I figured out that SQLite, which we use for our database, does not support this. @vanlummelhuizen was already investating a switch to MySQL for performance reasons, being able to set the collation (=way to sort characters) would be another reason to do it.
* Next I tried to use the sorted function in Python (so resorting after the database query), which has quite some options to sort the way you want... but apparently you cannot tell it to put special characters last.
* Then I tried to split the query set into two smaller sets: one with glosses that start with letters, and with glosses that don't, but apparently you cannot both (1) filter on the first characters AND (2) give multiple options what this first character can be.

In the end I found a working solution, which is a variant of the last approach; instead of a normal filter I now use a regex filter that splits the query set into two sets based on whether the first character is in the range `[a-zA-Z]` . It works, but I'm not sure what it does in terms of performance.
